android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/MainActivity.java

```diff
@@ -762,10 +762,15 @@ public void onClick(View v) {
           // Launch PlayStore to update Chrome
           try {
             Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse("market://details?id=com.android.chrome"));
-            startActivity(intent);
+            if (intent.resolveActivity(getPackageManager()) != null) {
+              startActivity(intent);
+            }
           } catch (android.content.ActivityNotFoundException e) {
             // Link to Chrome if user is not signed in to Play Store
-            startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse("https://play.google.com/store/apps/details?id=com.android.chrome")));
+            Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse("https://play.google.com/store/apps/details?id=com.android.chrome"));
+            if (intent.resolveActivity(getPackageManager()) != null) {
+              startActivity(intent);
+            }
```

Comment:
For this one, I'm fairly certain that inserting the `if` statement in the first section ought nullify the `catch` statement; instead, shouldn't the `catch` be turned into an `else`?

Reply:
Yes, you are right, Line 765 should have an `if`. I'm inclined to leave the catch there in case there are scenarios we are not covering with the resolveActivity test around the Play Store, but will tweak the rest accordingly.